storage/badger/procedure/index_test.go

```diff
@@ -0,0 +1,27 @@
+package procedure
+
+import (
+	"testing"
+
+	"github.com/dgraph-io/badger/v2"
+	"github.com/stretchr/testify/require"
+
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/utils/unittest"
+)
+
+func TestInsertRetrieveIndex(t *testing.T) {
+	unittest.RunWithBadgerDB(t, func(db *badger.DB) {
+		blockID := unittest.IdentifierFixture()
+		index := unittest.IndexFixture()
```

Comment:
⚠️ 
Here, you are testing only that the in-memory cache holds the provided item. (correct retrieval from the DB is not tested, as this comes into play only as a fallback when the cache do _not_ contain the requested entity. 

I believe if you extended your test similarly to  https://github.com/onflow/flow-go/blob/2375004eb5ecee3a16889403b1b75eea0832ee2b/storage/badger/blocks_test.go#L63-L69 the test should fail for any Index with `NodeIDs` ... well, this failure case is concealed by your revision: https://github.com/onflow/flow-go/blob/2375004eb5ecee3a16889403b1b75eea0832ee2b/utils/unittest/fixtures.go#L1061 where you essentially hard-code the

Reply:
I don't think we cache the index. 

But yes, I have added the test case in blocks_test to prevent any similar bug that caused by caching.

Reply:
🙇